core/src/main/java/com/linecorp/armeria/common/DefaultTimeoutController.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import javax.annotation.Nullable;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import io.netty.channel.EventLoop;
+
+/**
+ * A controller that schedules the timeout task with the initial value or reschedule when the timeout
+ * setting is changed.
+ *
+ * <p>Note: This interface is meant for internal use only.
+ */
+public final class DefaultTimeoutController implements TimeoutController {
+
+    private final TimeoutTask timeoutTask;
+    private final Supplier<? extends EventLoop> eventLoopSupplier;
```

Comment:
Seems like there's no need to use a `Supplier` because `EventLoop` is always available for all callers if I read correctly.